src/bin/common/solving_loop.ml

```diff
@@ -944,6 +931,19 @@ let main () =
           { solver_ctx with ctx = cnf }
         ) st
   in
+  let handle_stmts all_context st l =
+    let rec aux named_map st = function
+      | [] -> st
+      | [main_stmt] ->
+        let st = handle_stmt all_context st main_stmt in
+        State.set named_terms named_map st
+      | named_stmt :: tl ->
+        let st = handle_stmt all_context st named_stmt in
+        let named_map = get_named_of_stmt ~acc:named_map named_stmt in
```

Comment:
Should we maybe only do this if `named_stmt.implicit` is true? Although I think all terms but the last one are implicit currently that could change — I think it's best to dispatch on whether `implicit` is `true/false` for calling `get_named_of_stmt` rather than if it is the last term or not, might save us some surprises down the line (or not, but hey).